File: 4th semmester/oop/lista_2/java/ExString.java
// Julio Cesar Garcia Ribeiro - RA: 1994484

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class ExString {
    public static void main(String args[]) {
        InputStreamReader c = new InputStreamReader(System.in);
        BufferedReader cd = new BufferedReader(c);
        
        String frase = new String();
        char letra = '\0';
        int count = 0;
        
        System.out.println("Digite uma frase:");
        try {
            frase = cd.readLine();
        }
        catch(IOException e) {
            System.out.println("Erro de entrada");
            System.exit(1);
        }
        frase = frase.toLowerCase();
        
        System.out.println("Digite uma letra:");
        try {
            letra = (char)cd.read();
        }
        catch(IOException e) {
            System.out.println("Erro de entrada");
            System.exit(1);
        }
        letra = Character.toLowerCase(letra);
        
        for(int i = 0; i < frase.length(); i++) {
            if(Character.compare(frase.charAt(i), letra) == 0) {
                count++;
                System.out.println("Letra encontrada na posicao frase[" + i + "]");
            }
        }
        
        if(count == 0)
            System.out.println("Letra nao foi encontrada na frase");
        else
            System.out.println("A letra " + letra + " foi encontrada " + count + " vezes na frase");
    }
}

File: oop/lista_2/c/exString.c
// Julio Cesar Garcia Ribeiro - RA: 1994484

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[]) {
    char frase[50];
    char letra;
    int count = 0;

    printf("Digite uma frase: ");
    setbuf(stdin, NULL);
    fgets(frase, 49, stdin);
    frase[strcspn(frase, "\n")] = '\0';
    printf("Digite uma letra minuscula: ");
    setbuf(stdin, NULL);
    scanf("%c", &letra);

    if(letra < 97 || letra > 122) {
        printf("Letra invalida\n");
        exit(1);
    }

    for(int i = 0; i < strlen(frase); i++) {
        if ((frase[i] >= 65 && frase[i] <= 90) || (frase[i] >= 97 && frase[i] <= 122)) {
            if(letra == frase[i] || letra-32 == frase[i]) {
                printf("Letra encontrada na posicao frase[%d]\n", i);
                count++;
            }
        }
    }

    if(count == 0)
        printf("Letra nao foi encontrada na frase\n");
    else
        printf("A letra %c foi encontrada %d vezes na frase\n", letra, count);
}